### PR TITLE
Acceptance tests doc

### PR DESCRIPTION
**Pull request by lovromazgon**
### Description

This PR creates a document that gathers my research into acceptance tests. This was triggered by the integration of OpenCDC records into the SDK. It is a dump of my thoughts on acceptance tests that should serve as a basis for discussion around this topic. If/when we agree on the approach, we can take information from this document to create specific issues around implementing acceptance tests.

### CODE REVIEW

File: docs/acceptance-tests.md [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+    srcD1 -- No --> srcSignaling;
+    srcD1 -- Yes ---> srcD2;
+    srcD2 -- No --> srcAppending;
+    srcD2 -- Yes --> srcMutating;
+```
+
+All source tests are executed with structured records. The reason is that not all destinations are able to consume unstructured raw data, while structured data should be supported by any destination connector. We should give the user control over field names in the structured record in case the connector expects specific field names that match the target structure.
+
+| Test                                                 | ![Signaling source](https://img.shields.io/badge/-Signaling%20source-yellow?style=flat-square) | ![Appending source](https://img.shields.io/badge/-Appending%20source-yellowgreen?style=flat-square) | ![Mutating source](https://img.shields.io/badge/-Mutating%20source-green?style=flat-square) |
```
Comment (maha-hajja) [inline docs/acceptance-tests.md:47]: maybe add some info about what this table presents. and that you'll get into more details in the doc

File: docs/acceptance-tests.md [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+  1. Open source
+  1. Write X unrelated structured records to destination
+  1. Read all records from source
+  1. Expect X raw or structured records
+
+### Can the destination connector write raw records?
+
+Expected to pass:
+![Appending desstination](https://img.shields.io/badge/-Appending%20destination-yellowgreen?style=flat-square)
+![Mutating destination](https://img.shields.io/badge/-Mutating%20destination-green?style=flat-square)*
```
Comment (maha-hajja) [inline docs/acceptance-tests.md:256]: is the `*` here for a reason?

File: docs/acceptance-tests.md [hunk 3]
```diff
@@ -0,0 +1,298 @@
+# Acceptance tests
+
+This document describes one possible approach for writing acceptance tests for Conduit
+connectors. We discovered the need to revisit this when integrating the OpenCDC record
```
[the remark this answers is not in the document]
Comment (lovromazgon) [reply docs/acceptance-tests.md]: By "this" I mean acceptance tests. This document is the first step in that process.

File: docs/acceptance-tests.md (hunk 1 above)
Comment (lovromazgon) [reply docs/acceptance-tests.md:56]: I added a sentence describing the table.

**Review by maha-hajja — APPROVED** [review]
LGTM

**Pull request merged**